http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2StreamHandler.java

```diff
@@ -182,8 +174,10 @@ public void onHeaders(Stream stream, HeadersFrame frame) {
     if (frame.isEndStream()) {
       result.sampleEnd();
       result.setPendingResponse(false);
+      if (!result.isSync()) {
+        result.completeAsyncSample();
```

Comment:
consider putting sampleEnd, setPendingResponse(false) and if isSync + completeAsync in a completeSample method to reduce code duplication?